### PR TITLE
Implements GetKey

### PR DESCRIPTION
**Pull request by mmoadeli**
It also include some earlier work on tests

[Review on Reviewable](https://reviewable.io/reviews/dirvine/routing/166)

### CODE REVIEW

File: src/routing_node.rs [hunk 1]
```diff
@@ -692,6 +735,16 @@ impl<F> RoutingNode<F> where F: Interface {
         )
     }
 
+    fn construct_get_key_response_msg(&self, header : MessageHeader,
```
Comment (benjaminbollen) [inline src/routing_node.rs]: better to pass the original header, otherwise the `construct_get_key_response_msg()` function is a bit pointless

**Comment by dirvine** [comment]
Reviewed files:
- src/messages.rs @ r1
- src/messages/get_client_key.rs @ r1
- src/messages/get_client_key_response.rs @ r1
- src/messages/get_key.rs @ r1
- src/messages/get_key_response.rs @ r1
- src/node_interface.rs @ r1
- src/routing_node.rs @ r2
- src/sentinel.rs @ r1
- src/test_utils/messages_util.rs @ r1
- src/types.rs @ r1

---





Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/166)

**Comment by dirvine** [comment]
<sup>**[src/routing_node.rs, line 738 \[r2\]](https://reviewable.io:443/reviews/dirvine/routing/166#-JoA-0BYS9gwVYYwPRP8-r2-738)** ([raw file](https://github.com/dirvine/routing/blob/c7c8e55cc7f00d953467544848f9a3a295e8b7d8/src/routing_node.rs#L738)):</sup>
makes sense, what do you think Mahmoud?

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/166)

**Comment by mmoadeli** [comment]
<sup>**[src/routing_node.rs, line 738 \[r2\]](https://reviewable.io:443/reviews/dirvine/routing/166#-JoA-0BYS9gwVYYwPRP8-r2-738)** ([raw file](https://github.com/dirvine/routing/blob/c7c8e55cc7f00d953467544848f9a3a295e8b7d8/src/routing_node.rs#L738)):</sup>
David, initially it was a reference, but it failed as the RoutingMessage::new function does not take reference to header.  So, between clarity and performance, I chose the former.  Perhaps I should have gone for the latter, though.

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/166)

**Comment by dirvine** [comment]
<sup>**[src/routing_node.rs, line 738 \[r2\]](https://reviewable.io:443/reviews/dirvine/routing/166#-JoA-0BYS9gwVYYwPRP8-r2-738)** ([raw file](https://github.com/dirvine/routing/blob/c7c8e55cc7f00d953467544848f9a3a295e8b7d8/src/routing_node.rs#L738)):</sup>
OK makes sense, lets go with it for now in any case then. Thanks Mahmoud 

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/166)

**Comment by dirvine** [comment]
Reviewed files:
- src/routing_node.rs @ r3

---





Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/166)

**Comment by benjaminbollen** [comment]
This needed a version bump, changes node interface

**Pull request merged**